mlflow/tracking/fluent.py

```diff
@@ -2430,8 +2462,8 @@ def setup_autologging(module):
     # for each autolog library (except pyspark), register a post-import hook.
     # this way, we do not send any errors to the user until we know they are using the library.
     # the post-import hook also retroactively activates for previously-imported libraries.
-    for module in list(set(LIBRARY_TO_AUTOLOG_MODULE.keys()) - {"pyspark", "pyspark.ml"}):
-        register_post_import_hook(setup_autologging, module, overwrite=True)
+    for library in list(set(target_library_and_module) - {"pyspark", "pyspark.ml"}):
```

Comment:
```suggestion
    for library in sorted(set(target_library_and_module) - {"pyspark", "pyspark.ml"}):
```

to ensure the iteration order is always the same.